### DISCUSSION TITLE
CI/CD - Github Actions to build Vts.sln and run Vts.Library.sln Tests upon PR submission

### DISCUSSION THREAD

**Discussion by dcuccia** [VirtualPhotonics/VTS#87] [Ideas]
To dip our toe in CI, suggest we create an Action that automatically builds Vts.sln, and tests Vts.Library.sln as part of every PR. I can take this on.

**Comment by dcuccia** [comment]
Sorry - I prematurely entered this, but have now updated the title and primary entry ^.

**Comment by lmalenfant** [comment]
I created a POC on a fork and was able to implement the build and test workflow against Vts.sln. I deleted the VtsLibrary.sln because it was created to facilitate only running analysis on parts of the solution and it is no longer necessary.

https://github.com/lmalenfant/VTS/actions/workflows/dotnet.yml

This is the yaml file that was setup to run on all commits to a feature branch and on pull requests to master:
https://github.com/lmalenfant/VTS/blob/feature/lmalenfant-patch-1/.github/workflows/dotnet.yml

I think we should add a develop branch to our repo and run the action on push & PRs to develop and master only (after discovering that GitHub Actions are limited to 2000 minutes per month). 

I should mention that we have had a GitHub Action running on the Vts.MonteCarlo repository for the past 7 months, it does build, test and code analysis using SonarCloud. We have an issue with code coverage which we have been unable to resolve, any assistance that could be provided here would be useful.

https://github.com/VirtualPhotonics/Vts.MonteCarlo/actions

**Reply by lmalenfant** [reply]
I realized yesterday that I use the VtsLibrary.sln for the SonarCloud analysis that runs locally. I will see if I can make it work with the csproj so I can remove the extra solution file.

**Reply by lmalenfant** [reply]
I was finally able to get the code coverage working from the GitHub Action, the MCCL CI is working great now, I think we are ready to enable this for the VTS also.

**Reply by dcuccia** [reply]
Nice work!
